### DISCUSSION TITLE
Why is GPT-4 not working?

### DISCUSSION THREAD

**Discussion by dbmcco** [brianpetro/obsidian-smart-connections#147] [Q&A] [answered]
GPT3.5 is working fine, but 4 is throwing an error (I have the plus subscription).  Not sure what the error means though.

BTW - awesome progress on a really great implementation.

<img width="894" alt="Screenshot 2023-04-03 at 8 06 47 AM" src="https://user-images.githubusercontent.com/89109181/229505451-4da3f3de-4e73-4885-b34b-ff3dd1c8b45d.png">

<img width="369" alt="Screenshot 2023-04-03 at 8 06 15 AM" src="https://user-images.githubusercontent.com/89109181/229504799-86d8ac0d-e37c-4268-922a-08f7c84588b4.png">

**Comment by brianpetro** [comment, marked as answer]
Hi @dbmcco ,

Even with ChatGPT Plus, the GPT-4 API still requires special access. You can join the wait-list for GPT-4 at this link https://openai.com/waitlist/gpt-4-api

Brian 🌴

**Reply by dbmcco** [reply]
Ahh, of course - I should I realized!   Thanks - I just joined the wait list.  And thanks for the quick reply.

**Reply by jeremygentles** [reply]
@brianpetro I am having a similar issue. GPT-3.5 works but 4 does not. I DO have access to GPT-4 in the sandbox and have paid for plenty of credits. Any thoughts?

<img width="505" alt="Screenshot 2023-10-11 at 2 49 40 PM" src="https://github.com/brianpetro/obsidian-smart-connections/assets/6180613/fa5e9659-4cfa-4cdc-a0f7-e1ef142f9366">

<img width="311" alt="Screenshot 2023-10-11 at 2 52 10 PM" src="https://github.com/brianpetro/obsidian-smart-connections/assets/6180613/589f8a19-39a4-4436-a7da-f18a510345f5">

**Reply by brianpetro** [reply]
@jeremygentles hmmm. I'll have to think about this one. It should work if you see the model in the playground. 

Maybe just verify that you're using the API key that's associated with that account. But besides that, I'll have to think about it some more.

🌴

**Reply by jeremygentles** [reply]
@brianpetro I am using an API key associated with the/my account. Key is valid when API key is tested. Also generated a new key and still does not work. Not sure if logs from the developer console will help, but have included below. Let me know if you have any thoughts or need other details. 

 
<img width="640" alt="Screenshot 2023-10-12 at 8 44 18 AM" src="https://github.com/brianpetro/obsidian-smart-connections/assets/6180613/a7770d7f-cbd6-49d4-b440-4584603cd8ad">

**Reply by brianpetro** [reply]
@jeremygentles thanks for sharing those error logs, it is helpful.

For some reason, even though your account has GPT-4 in the playground, it doesn't have it via the API (see where it says "model not found").

Unfortunately, there's nothing else I can do from my end so I recommend reaching out to OpenAI support to see what they have to say about what's happening.

🌴

**Reply by jeremygentles** [reply]
thanks!

**Reply by bkriddle** [reply]
Did you figure out what the problem was?  I'm having the same issue.  Was it a problem on OpenAI's end?
